prow/bugzilla/client.go

```diff
@@ -75,25 +78,77 @@ type Client interface {
 	RemovePullRequestAsExternalBug(id int, org, repo string, num int) (bool, error)
 	// GetAllClones returns all the clones of the bug including itself
 	// Differs from GetClones as GetClones only gets the child clones which are one level lower
-	GetAllClones(bug *Bug) ([]*Bug, error)
+	// GetAllClones will also fetch bugDetails and the root bug details in addition to clones
+	// since we want to parallelize all the HTTP calls
+	GetAllClones(bugID int) (clones []*Bug, bug *Bug, root *Bug, err error)
 	// GetRootForClone returns the original bug.
 	GetRootForClone(bug *Bug) (*Bug, error)
+	// GetFromCache will fetch the clones from the clones cache
```

Comment:
Why do we need these two? I don't think the public interface for the client should allow for this sort of thing.

Reply:
I added the clones list cache to the client instead of keeping it as a package variable. The only way I could access it from the client interface was to add these functions to set and get from cache. Let me know if some other method is appropriate for this.